File: 3723/3723.c
int maximizeSumOfSquaresOfDigits(struct TreeNode* root) {
    if (!root) return 0;
    int l = maximizeSumOfSquaresOfDigits(root->left);
    int r = maximizeSumOfSquaresOfDigits(root->right);
    return 1 + (l > r ? l : r);
}

File: 3723/3723.go
func maximizeSumOfSquaresOfDigits(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximizeSumOfSquaresOfDigits(root.Left)
	r := maximizeSumOfSquaresOfDigits(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}